src/systems/day_night_cycle.rs

```diff
@@ -0,0 +1,64 @@
+use amethyst::{
+    core::{Time},
+    ecs::*,
+    audio::{SourceHandle},
+};
+
+use crate::Music;
+
+pub struct DayNightCycle{
+    pub cycle_time: f32,
+    pub elapsed_cycle_time: f32,
+    pub current_cycle: CycleState,
+    pub cycle_data: CycleData,
+}
+
+impl DayNightCycle {
+    pub fn new() -> Self {
+        DayNightCycle{
+            cycle_time: 6.0, // FIXME 1 second for testing
+            current_cycle : CycleState::Day,
+            elapsed_cycle_time: 0.0,
+            cycle_data: CycleData {
+                day_music_name: "day".into(),
+                night_music_name: "night".into(),
+            }
+        }
+    }
+}
+
+#[derive(Copy,Clone,Debug)]
+pub enum CycleState{
+    Day,
+    Night,
+}
+
+pub struct CycleData{
+    pub day_music_name: String,
+    pub night_music_name: String,
+}
+
+impl<'s> System<'s> for DayNightCycle{
```

Comment:
It might make sense to split this up into two systems. One that determines if it is day or night and one that switches the music based on that.
The system switching the music could be event driven. The system determining if it is day or night would dispatch an event with the `CycleState` as the payload.
The advantage of splitting that is that you'd be able to enable/disable individual systems for faster prototyping. In addition, other effects of the day/night cycle don't need to modify existing systems but can be built on top.

Reply:
Ok, thanks! I'll look into events :)